src/validation_workflow/docker/validation_docker.py

```diff
@@ -29,23 +26,56 @@ def __init__(self, args: ValidationArgs) -> None:
         super().__init__(args)
 
     def download_artifacts(self) -> bool:
-
-        # Check if the docker daemon is running; or else, bring it up
-        if (self.is_container_daemon_running()):
-
-            # STEP 1 . pull the images for OS and OSD
-            self._OS_image_name = 'opensearchproject/opensearch' if self.args.docker_source == 'dockerhub' else 'public.ecr.aws/opensearchproject/opensearch'
-            self._OSD_image_name = 'opensearchproject/opensearch-dashboards' if self.args.docker_source == 'dockerhub' else 'public.ecr.aws/opensearchproject/opensearch-dashboards'
-
-            self.local_image_OS_id = self.get_image_id(self._OS_image_name, self.args.version)
-            self.local_image_OSD_id = self.get_image_id(self._OSD_image_name, self.args.version)
+        try:
+            assert self.is_container_daemon_running(), 'Docker Daemon is not running. Exiting the docker validation.'
+
+            image_names = {
+                'dockerhub': {
+                    'os': {
+                        True: 'opensearchstaging/opensearch',
+                        False: 'opensearchproject/opensearch'
+                    },
+                    'osd': {
+                        True: 'opensearchstaging/opensearch-dashboards',
+                        False: 'opensearchproject/opensearch-dashboards'
+                    }
+                },
+                'ecr': {
+                    'os': {
+                        True: 'public.ecr.aws/opensearchstaging/opensearch',
+                        False: 'public.ecr.aws/opensearchproject/opensearch'
+                    },
+                    'osd': {
+                        True: 'public.ecr.aws/opensearchstaging/opensearch-dashboards',
+                        False: 'public.ecr.aws/opensearchproject/opensearch-dashboards'
```

Comment:
I did not get the True/False mapping. Can you explain? AFAI understand, its whether its staging or prod. Would ask to name the key of the mapping accordingly. True/False is confusing or add comments.
Adding @prudhvigodithi / @peterzhuamazon if they can suggest something better.

Reply:
thanks, @gaiksaya.. because we only have Staging and Production, and I want to determine which one to apply by one of the input argument, 'self.args.using_staging_artifact_only', which will be either True or False; hence, I wanted to map Staging = True, and Production = False. 
Therefore, at L56~57, I can apply to the correct URL based on the input argument 'self.args.using_staging_artifact_only'.

Reply:
Myabe we can change to something like this?